### PR TITLE
Fixed affiliated user domains usage for selection affiliated domains on entity add/edit form.

### PR DESCRIPTION
**Pull request by AntonAvator**
Fix for https://github.com/skilld-labs/domain_entity/issues/6 issue.

### CODE REVIEW

File: domain_entity.module [hunk 1]
```diff
@@ -491,3 +493,33 @@ function _domain_entity_get_related_domains(EntityInterface $entity) {
...
+  if ($context['items']->getFieldDefinition()->getName() !== DomainEntityMapper::FIELD_NAME) {
+    return;
+  }
+  if (\Drupal::currentUser()->hasPermission('publish to any domain')) {
+    return;
+  }
+
+  // Make affiliate domain selection required in case if user
+  // have no "publish to any domain" permission to avoid ambiguity.
+  $element['#required'] = TRUE;
```
Comment (andypost) [inline domain_entity.module]: I'd better write that as 
``` php
$element['#required'] = !\Drupal::currentUser()->hasPermission('publish to any domain');
```

File: domain_entity.module [hunk 2]
```diff
@@ -491,3 +493,33 @@ function _domain_entity_get_related_domains(EntityInterface $entity) {
...
+ * Implements hook_domain_references_alter().
+ */
+function domain_entity_domain_references_alter($query, $account, $context) {
+
+  if ($account->hasPermission('publish to any domain')) {
+    return;
+  }
+
+  $allowed = \Drupal::service('domain_access.manager')->getAccessValues($account);
+  $query->condition('id', array_keys($allowed), 'IN');
```
Comment (andypost) [inline domain_entity.module]: Please add code comment a kind of `// Limit list of domains to a set that user can use.`

**Review by andypost — CHANGES_REQUESTED** [review]
Please clean-up code a bit

File: domain_entity.module (hunk 1 above)
Comment (AntonAvator) [reply domain_entity.module]: Updated.

File: domain_entity.module (hunk 2 above)
Comment (AntonAvator) [reply domain_entity.module]: Comment was added.

**Pull request merged**